src/borg/compress.pyx

```diff
@@ -72,10 +72,14 @@ cdef class CompressorBase:
         using Auto compression it needs to determine the _actual_ target compression of a chunk
         in order to detect whether it should be recompressed.
 
-        For all Compressors that are not Auto this always returns *self*.
+        Any compressor may return a compressor other than *self*, like e.g. the CNONE compressor,
+        and should actually do so if *data* would become larger on compresson.
         """
         return self
 
+    def decide_compress(self, data):
+        raise NotImplementedError
```

Comment:
remove `decide` and `decide_compress` from `CompressorBase` and only have them in `DecidingCompressor` base class?

Reply:
Makes sense I guess, no guarantee for it to be available and thus not requiring it and leaving it up to its users to only use it with compressors supporting it.

Reply:
did you forget to remove the 2nd method or do you think it is needed?

Reply:
Both actually. :smile:  I did miss it reading your comment, but I also do think it is needed, it at least was in the base class (and thus any compressor) before any of my changes, and IIRC recreate does need to call decide() when recompressing, I'll check that.

Reply:
Yes, [`chunk_processor()`](https://github.com/borgbackup/borg/blob/27a41f5c1f8a8187c9ae129d1ec1de73e6a8712d/src/borg/archive.py#L1964) in class `ArchiveRecreater` does use `decide()` unconditionally, so it is needed.

Reply:
ah, ok.